tests/ops/op_math/test_sum.py

```diff
@@ -821,7 +820,7 @@ def circuit(weights):
         assert qnp.allclose(grad, true_grad)
 
     def test_non_hermitian_op_in_measurement_process(self):
-        """Test that non-hermitian ops in a measurement process will raise a warning."""
+        """Test that non-hermitian ops in a measurement process will raise an error."""
```

Comment:
Part of the reason we decided to go with a warning in the past was because our check for hermiticity is not an exhaustive check (i.e certain ops may pass the test and still be non-hermitian). 

I think if we raise an error when the test fails this might be setting a double standard since we are already consciously allowing non hermitian ops to get measure silently if they pass the incomplete check. 

Worth having a chat about before we swap this back. Is it blocking something?

Reply:
The error was raised because `Sum(qml.RX(1.23, wires=0), qml.Identity(wires=1))` was separated into two different tapes, and `DefaultQubit` doesn't support `qml.RX` as observable.

I changed the test such that the `non-hermitian` warning is raised.

Reply:
Awesome thanks !